libnopegl/src/gen_specs.c

```diff
@@ -288,6 +288,11 @@ static int check_node_params(const struct node_class *cls)
     if (!par)
         return 0;
 
+    if (cls->params && !cls->opts_size) {
```

Comment:
nit: 
- "swapped inconsistency" sounds confusing, maybe something like: The check for "no params but an opts size" is [...] and the scenario would only lead to unused memory to be allocated
- while forgetting to set the options size would lead to undefined behavior and probably a crash

Reply:
Reworded that way:
```
gen_specs: add a params + opts_size consistency check

The check for no params but an opts size is unnecessary because the
compiler would spot the issue (unused parameters array) and only leads
to an unused allocated memory, while forgetting to set the options size
is silent at compilation and leads to an obscure crash.
```